### DISCUSSION TITLE
Enchancement

### DISCUSSION THREAD

**Discussion by object-Object** [HexxyCraft/modpack#8] [General]
Should we include the Enchancement mod, and if so, how should we configure it?

**Comment by beholderface** [comment]
Personally, I don't think it would add very much value to the pack. There were people on discord mentioning how disabling durability would reduce the problem of "everyone only uses Botania's mana-autorepair tools", but with the current modlist there aren't that many alternative tool options anyway.
Then there's the issue of it breaking mods that use durability for things that aren't just "the tool will break eventually if you don't take care of it".
Some of the enchantments it adds look neat, but I'd rather have mostly-vanilla enchantment mechanics than cool modded enchants and questionable sweeping changes

**Comment by maraxin** [comment]
> There were people on discord mentioning how disabling durability would reduce the problem of "everyone only uses Botania's mana-autorepair tools", but with the current modlist there aren't that many alternative tool options anyway.

Mending exists

> Then there's the issue of it breaking mods that use durability for things that aren't just "the tool will break eventually if you don't take care of it".

Which mod will have this problem? There are no mods, that would be really broken because of this

**Comment by object-Object** [comment]
The issue is we won't really know what breaks until we find broken things. And if there are no current issues and we add it, then we later add a mod that does break, that's hard to resolve.

**Comment by LlamaSageR** [comment]
Enchancement should not be added. The features it adds are not worth the invasive changes it makes. Here's a list:

- Removal of nigh every enchantment. The only ones preserved from Vanilla are Fire Aspect, Infinity, Channeling, Riptide, Luck of the Sea, Efficiency, and Silk Touch. The better ore procurement of Fortune is gone, the only damage increase for swords is now *fire aspect* and the Enchancement berserk effect, every form of additional protection on armor is removed, and the underwater utility of Aqua Affinity, Respiration, and Depth Strider is absent. Not to mention that other mods add enchantments, and catching them all would be difficult (In the discussion on Discord, no one even mentioned the enchantment Supplementaries provides). The sheer amount of utility vanilla enchants provides is missing, and reintroducing them would be pointless due to the other changes Enchancement provides, explained next.

- Enchantment limit of 1 per item. This makes sense in the native Enchancement ecosystem, since each enchantment is a unique choice. However, it entirely prevents the reintroduction of other enchantments, because many enchantments are *supplemental* to the gear, rather than a major trait. Why add Aqua Affinity without Respiration? And why choose either of those over a stronger enchantment like Protection (for example; Protection was removed as well). Rather than encouraging choice, it is stifling. And increasing the limit isn't an option either, since the majority of the introduced enchantments are not designed to work together! While they can in principle, it's always supposed to be an exclusive choice.

- The removal of enchantment leveling. Every enchantment is level 1. While they do tweak the values to be the same as default (the new Riptide I is the same as the old Riptide III), this makes integrating other enchantments impossible. Fortune I is useless compared to Fortune III. This also prevents rolling back Enchancement, since every single item will have only 1 level per enchantment.

- The removal of durability from the game. Here's the big one. The removal of Durability absolutely kills mod compatibility, since many systems are designed with Durability in mind. Why sacrifice your chest slot for the Copper Backtank to mitigate the durability damage on the Extend-o-grip and Potato Cannon? Why burn through pools of mana to refill your Terrasteel armor when it takes no damage at all? It also entirely removes the point of repairing and other systems, which can be fixed and already have been in the modpack via mods like Charm and Tinkerer's Smithery. And that's not even accounting for the possible errors that could crop up 50 hours in that lead to broken interactions! A lot of mods use durability even if it doesn't look like they do, and we cannot know until we break it ourselves. And once it does break? *There's nothing we can do because the mod is so invasive.*

While it is possible to config the mod, the systemic problems will never go away. Not to mention that the enchantments aren't really that good. The mobility is neat, but not at the expense of literally everything (Not to mention it invalidates innovative tools from Botania and Hex Casting). The only way to fix the mod would be to remove nearly all of it and keep configuring it as the modpack evolves, which is definitely not worth the effort when there are easier solutions.

This is why Enchancements should not be added.

**Comment by maraxin** [comment]
Wall of text. Won't answer, but you have some mistakes here, LlamaSageR

**Comment by object-Object** [comment]
If you disagree with points they made, comment on them or forever hold your peace, because as of right now I'm not seeing any compelling reasons to include the mod in the pack.

**Comment by LlamaSageR** [comment]
A comprehensive problem requires a comprehensive answer. To shorten it would be to sacrifice needed information. If you have a counterargument or other critique, please send it.
